flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/LaunchableMesosWorker.java

```diff
@@ -262,9 +264,61 @@ public String toString() {
 			taskInfo.setContainer(containerInfo);
 		}
 
+		containerInfo.addAllVolumes(buildVolumes(params.containerVolumes()));
+
 		return taskInfo.build();
 	}
 
+	/**
+	 * Used to build volume specs for mesos. This allows for mounting additional volumes into a container
+	 *
+	 * @param containerVolumes a comma delimited optional string of [host_path:]container_path[:RO|RW] that
+	 *                         defines mount points for a container volume. If None or empty string, returns
+	 *                         an empty iterator
+	 */
+	public static List<Protos.Volume> buildVolumes(Option<String> containerVolumes) {
+		if (containerVolumes.isEmpty()) {
+			return new ArrayList<Protos.Volume>();
+		}
+		String[] specs = containerVolumes.get().split(",");
+		List<Protos.Volume> vols = new ArrayList<Protos.Volume>();
+		for (String s : specs) {
+			if (s.trim().isEmpty()) {
+				continue;
+			}
+			Protos.Volume.Builder vol = Protos.Volume.newBuilder();
+			vol.setMode(Protos.Volume.Mode.RW);
+
+			String[] parts = s.split(":");
+			switch (parts.length) {
+				case 1:
+					vol.setContainerPath(parts[0]);
+					break;
+				case 2:
+					try {
+						Protos.Volume.Mode mode = Protos.Volume.Mode.valueOf(parts[1].trim().toUpperCase());
+						vol.setMode(mode)
+							.setContainerPath(parts[0]);
+					} catch (IllegalArgumentException e) {
+						vol.setHostPath(parts[0])
+							.setContainerPath(parts[1]);
+					}
+					break;
+				case 3:
+					Protos.Volume.Mode mode = Protos.Volume.Mode.valueOf(parts[2].trim().toUpperCase());
+					vol.setMode(mode)
+						.setHostPath(parts[0])
+						.setContainerPath(parts[1]);
+					break;
+				default:
+					throw new IllegalArgumentException("volume specification is invalid, given: " + s);
```

Comment:
Ideally this validation would occur eagerly, i.e. in `MesosTaskManagerParameters::create`.  Flink emphasizes eager validation.